protocol/mqtt_paho/v2/protocol.go

```diff
@@ -0,0 +1,134 @@
+package mqtt_paho
```

Comment:
This file has no tests, can we get some coverage here e.g., using integration tests?

Reply:
I had planned to add an integration test in future PR, which might need to set some environment in the workflow.
If you think it's better to put it here, I will and these tests tomorrow~

Reply:
@embano1 I added some integration tests today, PTAL~